cmd/embedded-cluster/reset.go

```diff
@@ -116,14 +116,33 @@ func (h *hostInfo) checkQuorumSafety(c *cli.Context) (bool, string, error) {
 	if err != nil {
 		return false, "", fmt.Errorf("unable to unmarshal etcd member list, %w, %s", err, out)
 	}
+	// get a rough picture of the cluster topology
+	workers := []string{}
+	controllers := []string{}
+	nodeList := corev1.NodeList{}
+	err = h.Kclient.List(c.Context, &nodeList)
+	if err != nil {
+		return false, "", fmt.Errorf("unable to create kubernetes client: %w", err)
+	}
+	for _, node := range nodeList.Items {
+		labels := node.GetLabels()
+		if labels["node-role.kubernetes.io/control-plane"] == "true" {
+			controllers = append(controllers, node.Name)
+		} else {
+			workers = append(workers, node.Name)
+		}
+	}
+	if len(workers) > 0 && len(controllers) == 1 {
+		return false, "Cluster has a worker node joined and you are attempting to remove the only controller node.", nil
```

Comment:
A few thoughts.
* From an end-user perspective, controller is a node role, but the vendor might have [customized its name](https://docs.google.com/document/d/1NKXIj7fPhYc6BadscPznXMcbXVj74tBfY-xbNJqbVGk/edit#heading=h.bqi1aw166qh7). Can we show the actual node role name here instead of saying controller?
* Do you think we should tell them they can reset the workers first if they really want to reset this node, or should we just tell them the issue?
```suggestion
		return false, "Cannot reset the last controller node when there are other nodes in the cluster.", nil
```